subsys/bluetooth/host/classic/l2cap_br.c

```diff
@@ -1190,6 +1200,116 @@ static uint16_t l2cap_br_conf_opt_mtu(struct bt_l2cap_chan *chan,
 	return result;
 }
 
+static uint16_t l2cap_br_conf_opt_flush_timeout(struct bt_l2cap_chan *chan,
+				   struct net_buf *buf, size_t len)
+{
+	uint16_t result = BT_L2CAP_CONF_SUCCESS;
+	struct bt_l2cap_conf_opt_flush_timeout *opt_to;
+
+	if (len != sizeof(*opt_to)) {
+		LOG_ERR("qos frame length %zu invalid", len);
+		result = BT_L2CAP_CONF_REJECT;
+		goto done;
+	}
+
+	opt_to = (struct bt_l2cap_conf_opt_flush_timeout *)buf->data;
+
+	LOG_DBG("Flash timeout %u", opt_to->timeout);
+
+	opt_to->timeout = sys_cpu_to_le16(0xFFFF);
+	result = BT_L2CAP_CONF_UNACCEPT;
+done:
+	return result;
+}
+
+static uint16_t l2cap_br_conf_opt_qos(struct bt_l2cap_chan *chan,
+				   struct net_buf *buf, size_t len)
+{
+	uint16_t result = BT_L2CAP_CONF_SUCCESS;
+	struct bt_l2cap_conf_opt_qos *opt_qos;
+
+	if (len != sizeof(*opt_qos)) {
+		LOG_ERR("qos frame length %zu invalid", len);
+		result = BT_L2CAP_CONF_REJECT;
+		goto done;
+	}
+
+	opt_qos = (struct bt_l2cap_conf_opt_qos *)buf->data;
+
+	LOG_DBG("QOS Type %u", opt_qos->service_type);
+
+	if (opt_qos->service_type == BT_L2CAP_QOS_TYPE_GUARANTEED) {
+		/* Set to default value */
+		result = BT_L2CAP_CONF_UNACCEPT;
+		opt_qos->flags = 0x00;
+		/* do not care */
+		opt_qos->token_rate = sys_cpu_to_le32(0x00000000);
+		/* no token bucket is needed */
+		opt_qos->token_bucket_size = sys_cpu_to_le32(0x00000000);
+		/* do not care */
+		opt_qos->peak_bandwidth = sys_cpu_to_le32(0x00000000);
+		/* do not care */
+		opt_qos->latency = sys_cpu_to_le32(0xFFFFFFFF);
+		/* do not care */
+		opt_qos->delay_variation = sys_cpu_to_le32(0xFFFFFFFF);
+	}
+
+done:
+	return result;
+}
+
+static uint16_t l2cap_br_conf_opt_ret_fc(struct bt_l2cap_chan *chan,
+				   struct net_buf *buf, size_t len)
+{
+	uint16_t result = BT_L2CAP_CONF_SUCCESS;
+	struct bt_l2cap_conf_opt_ret_fc *opt_ret_fc;
+
+	if (len != sizeof(*opt_ret_fc)) {
+		LOG_ERR("ret_fc frame length %zu invalid", len);
+		result = BT_L2CAP_CONF_REJECT;
+		goto done;
+	}
+
+	opt_ret_fc = (struct bt_l2cap_conf_opt_ret_fc *)buf->data;
+
+	LOG_DBG("ret_fc mode %u", opt_ret_fc->mode);
+
+	if (opt_ret_fc->mode != BT_L2CAP_RET_FC_MODE_BASIC) {
+		/* Set to default value */
+		result = BT_L2CAP_CONF_UNACCEPT;
+		opt_ret_fc->mode = BT_L2CAP_RET_FC_MODE_BASIC;
+	}
+
+done:
+	return result;
+}
+
+static uint16_t l2cap_br_conf_opt_fcs(struct bt_l2cap_chan *chan,
+				   struct net_buf *buf, size_t len)
```

Comment:
And here. In many cases you can just have everything on one line.

Reply:
Updated.